### PR TITLE
Remove value field from connector default RCFs (batch 1)

### PR DESCRIPTION
**Pull request by navarone-feekery**
## Related to https://github.com/elastic/connectors-python/issues/606

Remove `value` field from connector default RCFs, allowing them to be initialised as a type-safe empty value.

- mongo
- mssql
- postgresql
- oracle
- mysql

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes
- [x] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/sources/azure_blob_storage.py [hunk 1]
```diff
@@ -105,7 +106,7 @@ def get_default_configuration(cls):
                 "validations": [
                     {"type": "less_than", "constraint": MAX_CONCURRENT_DOWNLOADS + 1}
                 ],
-                "value": MAX_CONCURRENT_DOWNLOADS,
+                "value": None,
```
Comment (tarekziade) [inline connectors/sources/azure_blob_storage.py:109]: if we always have to set `value` to `None`, what if we just remove the `value` key in that case for the default config -- seems to just be noise at that point or am I missing something ?

Comment (navarone-feekery) [reply connectors/sources/azure_blob_storage.py:109]: That's what I'll be doing! This PR is only complete for the DB ones.
(Also this is a draft please don't review it yet haha)

**Comment by artem-shelkovnikov** [comment]
@navarone-feekery should we close this PR? Should we still do the changes in it?

**Comment by navarone-feekery** [comment]
@artem-shelkovnikov I still want to do these changes but I didn't have time this release to finish it, can we hang onto the PR for one more release?
If I haven't finished it by then we can assume it isn't happening.

**Comment by navarone-feekery** [comment]
Memo: I need to merge [a prerequisite PR](https://github.com/elastic/connectors-python/pull/1323) first before I can continue on this current PR.

**Comment by artem-shelkovnikov** [comment]
> can we hang onto the PR for one more release?

Of course! Just checking if it's relevant work, since it's relevant we can keep it for now

**Review by seanstory — APPROVED** [review]
LGTM.

Does this need a corresponding kibana PR to remove all the `value` values for native connectors?

**Comment by navarone-feekery** [comment]
@seanstory I will take a look at the native connectors at the end of this, but at a glance they initialise with type-safe empty values already. Having the `value` field is okay, just trying to remove junk data.

**Pull request merged**